x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/action/SamlGenerateMetadataRequest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.idp.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class SamlGenerateMetadataRequest extends ActionRequest {
+
+    private String spEntityId;
+
+    public SamlGenerateMetadataRequest(StreamInput in) throws IOException {
+        super(in);
+        spEntityId = in.readString();
+    }
+
+    public SamlGenerateMetadataRequest(String spEntityId) {
+        this.spEntityId = Objects.requireNonNull(spEntityId, "Service Provider entity id must be provided");
+    }
+
+    public SamlGenerateMetadataRequest() {
+        this.spEntityId = null;
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        return null;
+    }
```

Comment:
There's no need to override this if it doesn't have a body.

Reply:
But `ActionRequest#validate` is abstract so I don't really have a choice :/

Reply:
Ah, so it is. Sorry, I forgot that was the case.